### ISSUE TITLE
Makefile.am misses to describe how to use DEFS

### ISSUE THREAD

**Issue by GitMensch** [westes/flex#333]
It says:

```
# Notes on building:

# Possible values for DEFS:
#
# By default, flex generates 8-bit scanners when using table compression,
# and 7-bit scanners when using uncompressed tables (-f or -F options).
# For flex to always generate 8-bit scanners, add "-DDEFAULT_CSIZE=256"
# to DEFS.
```

But doesn't says where/how to use it. I *assume* it means that a user may opt to specify this during `configure`: `./configure DEFS="-DDEFAULT_CSIZE=256"`, correct?

If yes, would you accept a PR with using this explanation there and more importantly in `configure.ac`?
If this is relevant only during configure I'd actually move the comment from Makefile.am to a help string in configure.ac.

**Comment by Explorer09** [comment]
Technically you can specify `DEFS` either in `./configure` or in `make` command, depending on your need.
Personally I won't bother to specify it in the readme or in `configure`, because such builds (about tweaking default behaviors) are quite specialized and not very useful for ordinary users (or distros).

**Comment by GitMensch** [comment]
In this case I opt for removing it seeing it as more confusing than helping (or move to configure.ac as a user helpstring). I'm available to create a PR for both... Postponed until comment from @westes.

**Comment by westes** [comment]
I'd rather see this in configure.ac if we're going to move it at all.

Or what would likely be better is to make sure the manual sufficiently documents the flags.

**Comment by westes** [comment]
There isn't a section in the manual on how to build flex.

There should be, however.

What I would suggest in the meantime is to create a section under the frequently asked questions and document it there. I'll get all that stuff moved out to the manual proper. (Perhaps not until 2.7.0 is seriously on the horizon, but having a section to move helps a bunch.)
